src/main/java/ru/skypro/homework/entity/User.java

```diff
@@ -0,0 +1,75 @@
+package ru.skypro.homework.entity;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NoArgsConstructor;
+import org.springframework.security.core.GrantedAuthority;
+import org.springframework.security.core.userdetails.UserDetails;
+import ru.skypro.homework.dto.Role;
+
+import javax.persistence.*;
+import javax.validation.constraints.Max;
+import javax.validation.constraints.Min;
+import java.util.Collection;
+import java.util.List;
+
+@Table(name = "users")
+@Entity
+@Data
+@NoArgsConstructor
+@AllArgsConstructor
+public class User implements UserDetails {
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    @Column(name = "id")
+    private Integer pk;
+    @Column(name = "email")
+    private String email;
+    @Column(name = "firstname")
+    private String firstName;
+    @Column(name = "lastname")
+    private String lastName;
+    @Column(name = "phone")
+    private String phone;
+    @Enumerated(EnumType.STRING)
+    @Column(name = "role")
+    private Role role;
+    @Column(name = "image")
+    private String image;
+    @Min(value = 8, message = "minLength: 8")
+    @Max(value = 16, message = "maxLength: 16")
+    @Column(name = "password")
+    private String password;
+    @OneToMany(mappedBy = "user")
+    private List<Ad> ads;
+
+    @Override
+    public Collection<? extends GrantedAuthority> getAuthorities() {
+        return null;
+    }
```

Comment:
Этот метод нужно реализовать. Примерно вот так:
```suggestion
    @Override
    public Collection<? extends GrantedAuthority> getAuthorities() {
        return Collections.singleton(new SimpleGrantedAuthority("ROLE_" + getRole()));
    }
```